app/controllers/spree/omniauth_callbacks_controller.rb

```diff
@@ -21,7 +21,7 @@ def provides_callback_for(*providers)
   def omniauth_callback
     if request.env['omniauth.error'].present?
       flash[:error] = I18n.t('devise.omniauth_callbacks.failure', kind: auth_hash['provider'], reason: I18n.t('spree.user_was_not_valid'))
-      redirect_back_or_default(root_url)
+      redirect_to stored_spree_user_location_or(root_url)
```

Comment:
This assumes we are using the latest version of solidus_auth_devise, which is not always true. We should probably make it conditional, isn't it?

Reply:
We should always be using the latest solidus_auth_devise as the version number is not specified in the [gemspec](  spec.add_dependency 'solidus_auth_devise') and it should resolve to the latest I would expect. It is true this would break if the user specified solidus_auth_devise 2.5.4, but I feel like we may want to specify in the gemspec rather than create a conditional here. WDYT?